helix-view/src/register.rs

```diff
@@ -20,28 +21,25 @@ use crate::{
 /// * Document path (`%`): filename of the current buffer
 /// * System clipboard (`*`)
 /// * Primary clipboard (`+`)
-#[derive(Debug)]
 pub struct Registers {
     /// The mapping of register to values.
     /// Values are stored in reverse order when inserted with `Registers::write`.
     /// The order is reversed again in `Registers::read`. This allows us to
     /// efficiently prepend new values in `Registers::push`.
     inner: HashMap<char, Vec<String>>,
-    clipboard_provider: Box<dyn ClipboardProvider>,
+    pub clipboard_provider: Box<dyn DynAccess<ClipboardProvider>>,
```

Comment:
I think this field can be made private again
```suggestion
    clipboard_provider: Box<dyn DynAccess<ClipboardProvider>>,
```